zerver/views/message_summary.py

```diff
@@ -29,6 +30,13 @@ def get_messages_summary(
     if not (user_profile.is_moderator or user_profile.is_realm_admin):  # nocoverage
         return json_success(request, {"summary": "Feature limited to moderators for now."})
 
+    if settings.MAX_CREDIT_USAGE_PER_MONTH is not None:
+        used_credits = (
+            COUNT_STATS["ai_credit_usage::day"].current_month_accumulated_count() / 1000000000
+        )
+        if used_credits >= settings.MAX_CREDIT_USAGE_PER_MONTH:
+            return json_success(request, {"summary": "Reached monthly limit for AI credits."})
```

Comment:
We probably will want to rework this so that it's a `json_error`; the client should be able to distinguish errors from success responses.

Reply:
Changed.